fitbenchmarking/core/tests/test_results_fallback.py

```diff
@@ -0,0 +1,660 @@
+                              {'row': 'prob_0', 'col': 's0:m01:j0:',
+                               'cat': 'cf1', 'result_ix': 2},
+                              {'row': 'prob_0', 'col': 's1:m10:j1:',
+                               'cat': 'cf1', 'result_ix': 4},
+                              {'row': 'prob_0', 'col': 's1:m11:j1:',
+                               'cat': 'cf1', 'result_ix': 5},
+                              {'row': 'prob_0', 'col': 's0:m01:j1:',
+                               'cat': 'cf1', 'result_ix': 6},
+                              {'row': 'prob_1', 'col': 's1:m10:j0:',
+                               'cat': 'cf1', 'result_ix': 7},
+                              {'row': 'prob_1', 'col': 's1:m11:j0:',
+                               'cat': 'cf1', 'result_ix': 8},
+                              {'row': 'prob_1', 'col': 's0:m01:j0:',
+                               'cat': 'cf1', 'result_ix': 9},
+                              {'row': 'prob_1', 'col': 's0:m00:j0:',
+                               'cat': 'cf1', 'result_ix': 10},
+                              {'row': 'prob_1', 'col': 's1:m10:j1:',
+                               'cat': 'cf1', 'result_ix': 11},
+                              {'row': 'prob_1', 'col': 's1:m11:j1:',
+                               'cat': 'cf1', 'result_ix': 12},
+                              {'row': 'prob_1', 'col': 's0:m01:j1:',
+                               'cat': 'cf1', 'result_ix': 13},
+                              {'row': 'prob_1', 'col': 's0:m00:j1:',
+                               'cat': 'cf1', 'result_ix': 14}]
+        self.expected_repeating_tags = []
+
+    def test_get_all_result_tags(self):
+        """
+        Test the _get_all_result_tags function
+        """
+        results = self.results
+        expected_tags = self.expected_tags
+        expected_repeating_tags = self.expected_repeating_tags
+
+        actual_tags, actual_repeating_tags = \
+            _get_all_result_tags(results,
+                                 self.sort_order,
+                                 self.col_sections)
+
+        self.assertEqual(actual_tags, expected_tags)
+        self.assertEqual(actual_repeating_tags, expected_repeating_tags)
+
+    def run_and_match_find_non_full_columns(self,
+                                            columns,
+                                            expected_count,
+                                            columns_with_errors,
+                                            expected_column_tags):
+        """
+        Helper function to call _find_non_full_columns
+        and match outputs
+        """
+        actual_column_tags = _find_non_full_columns(columns,
+                                                    expected_count,
+                                                    columns_with_errors)
+        self.assertEqual(actual_column_tags, expected_column_tags)
+
+    def test_find_non_full_columns(self):
```

Comment:
Can you put these in subtests.
Usually we have 1 class per function and several tests per class.

E.g.
```
class TestNonFullColumns:
    ...
    def test_all_columns_full(self):
        ...
    def test_some_full_columns(self):
        ...
    def no_full_columns(self):
        ...
```

But I think this with subtests is equivalent.
```
def test_non_full_columns(self):
    ...
    with self.subTest('all_columns_full'):
        ...
    ...
```